src/pip/_internal/commands/freeze.py

```diff
@@ -88,5 +88,8 @@ def run(self, options, args):
             skip=skip,
             exclude_editable=options.exclude_editable)
 
-        for line in freeze(**freeze_kwargs):
-            sys.stdout.write(line + '\n')
+        try:
+            for line in freeze(**freeze_kwargs):
+                sys.stdout.write(line + '\n')
+        finally:
+            wheel_cache.cleanup()
```

Comment:
Style nit. I'm not very keen on the fact that the try...finally is separated from the creation of the cache. Clearly there's no way the intermediate lines could fail, so in practice it's not a problem, but it looks odd (I went and looked at the source code to verify that the cleanup was in the right place).

This is of course one place where caches being context managers would make things clearer - so I'm OK with leaving this for tidying up when we look at doing that (and I agree with your comment that making that change should be a separate PR).